pkg/cloudprovider/aws/ami.go

```diff
@@ -48,7 +50,13 @@ func (p *AMIProvider) Get(ctx context.Context, constraints *Constraints) (string
 	if err != nil {
 		return "", fmt.Errorf("kube server version, %w", err)
 	}
-	name := fmt.Sprintf("/aws/service/bottlerocket/aws-k8s-%s/%s/latest/image_id", version, KubeToAWSArchitectures[*constraints.Architecture])
+	var architectureSuffix bytes.Buffer
+	if *constraints.Architecture == v1alpha3.ArchitectureArm64 {
+		architectureSuffix.WriteString("-arm64")
```

Comment:
why not just use a string here rather than a byte buffer?

Reply:
fixed